services/src/main/java/com/linkedin/kmf/services/configs/ProduceServiceConfig.java

```diff
@@ -122,7 +129,12 @@
                                     2.0,
                                     atLeast(1.0),
                                     ConfigDef.Importance.LOW,
-                                    CommonServiceConfig.PARTITIONS_TO_BROKER_RATIO_DOC);
+                                    CommonServiceConfig.PARTITIONS_TO_BROKER_RATIO_DOC)
+                            .define(TOPIC_FACTORY_CONFIG,
```

Comment:
To be consistent with the config `consme.consumer.class`, can we use ConfigDef.Type.STRING as type and DefaultTopicFactory.class.getCanonicalName() as default value?

One of the reason `consume.consumer.class` is of type STRING is because user can simply specify OldConsumer and NewConsumer as config value instead of specifying the full class path, which seems slightly easier for user. This can be user here since in the future we may have more default factory class in kafka-monitor.